portfolio/src/main/java/com/google/sps/data/LongestTaskFirstScheduler.java

```diff
@@ -101,4 +113,69 @@ public SchedulingAlgorithmType getSchedulingAlgorithmType() {
 
     return scheduledTimeRangeOptional;
   }
+
+  /**
+   * Keeps splitting up the task to schedule the currently avaible free time ranges until all of the
+   * task is scheduled across the different time ranges. When this method is called, the task may or
+   * may not be complete scheduled.
+   *
+   * @return a boolean value representing whether or not the task is complete scheduled.
+   */
+  private boolean splitUpTaskToSchedule(
+      Task task, TimeRangeGroup availableTimesGroup, List<ScheduledTask> scheduledTasks) {
+    List<TimeRange> currentAvailableTimes = constructAvailableTimeRanges(availableTimesGroup);
+    Duration taskDuration = task.getDuration();
+    int taskSegmentCount = 1;
+
+    for (TimeRange currentFreeTimeRange : currentAvailableTimes) {
+      if (taskDuration.getSeconds() > 0) {
+        Instant scheduledTime = currentFreeTimeRange.start();
+
+        // Reconstruct the task segment's name.
+        String taskName = task.getName() + " (Part " + taskSegmentCount + ")";
+        String taskDescription = "";
+        if (task.getDescription().isPresent()) {
```

Comment:
We can do

String taskDescription = task.getDescription().orElse("");

Reply:
Updated.